test/unit/org/apache/cassandra/cql3/RandomSchemaTest.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.cassandra.config.CassandraRelevantProperties;
+import org.apache.cassandra.db.marshal.DurationType;
+import org.apache.cassandra.db.marshal.UserType;
+import org.apache.cassandra.schema.ColumnMetadata;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.utils.AbstractTypeGenerators;
+import org.apache.cassandra.utils.AbstractTypeGenerators.TypeGenBuilder;
+import org.apache.cassandra.utils.CassandraGenerators;
+import org.apache.cassandra.utils.CassandraGenerators.TableMetadataBuilder;
+import org.apache.cassandra.utils.FailingConsumer;
+import org.quicktheories.core.Gen;
+import org.quicktheories.core.RandomnessSource;
+import org.quicktheories.generators.SourceDSL;
+import org.quicktheories.impl.JavaRandom;
+
+public class RandomSchemaTest extends CQLTester.InMemory
+{
+    private static final Logger logger = LoggerFactory.getLogger(RandomSchemaTest.class);
+
+    static
+    {
+        // make sure blob is always the same
+        CassandraRelevantProperties.TEST_BLOB_SHARED_SEED.setInt(42);
+    }
+
+    @Test
+    public void test()
+    {
+        Gen<Boolean> nulls = SourceDSL.integers().between(1, 100).map(i -> i < 5);
+        qt().checkAssert(random -> {
+            TypeGenBuilder nonEmptyNoDuration = AbstractTypeGenerators.builder()
+                                                                      .withoutEmpty()
+                                                                      .withUserTypeKeyspace(KEYSPACE)
+                                                                      .withoutPrimitive(DurationType.instance);
```

Comment:
`DurationType` violates several properties and is banned in several places do to these bugs... so exclude from this test as this test makes sure types respect their given properties